server/conn.go

```diff
@@ -997,12 +1009,71 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 			}
 		}
 	}
-	return errors.Trace(cc.writeEOF(more))
+	return errors.Trace(cc.writeEOF(serverStatus))
+}
+
+// writeChunksWithFetchSize writes data from a Chunk, which filled data by a ResultSet, into a connection.
+// binary specifies the way to dump data. It throws any error while dumping data.
+// serverStatus, a flag bit represents server information.
+// fetchSize, the desired number of rows to be fetched each time when client uses cursor.
+func (cc *clientConn) writeChunksWithFetchSize(ctx context.Context, rs ResultSet, serverStatus uint16, fetchSize int) error {
+	fetchedRows := rs.GetFetchedRows()
+
+	// if fetchedRows is not enough, getting data from recordSet.
+	for len(fetchedRows) < fetchSize {
+		chk := rs.NewChunk()
```

Comment:
I feel that this chunk cannot be reused. Because rows stored in fetchedRows refer to the underlying chunk.
If the chunk is reset, the data in Row is wrong.

Reply:
you are right~